pkg/api/v1beta1/clustercurator_types.go

```diff
@@ -45,6 +45,13 @@ type Hook struct {
 	// +kubebuilder:validation:Required
 	Name string `json:"name"`
 
+	// Type of the Hook. For Job type, Ansible job template will be used.
+	// For Workflow type, Ansible workflow template will be used.
+	// If omitted, default to Job type.
+	// +optional
+	// +kubebuilder:default=Job
+	Type HookType `json:"type,omitempty"`
```

Comment:
Is there a reason you called it hook and not AnsibleJobType or something?

Reply:
Did the AnsibleJob version change? Does the curator version need to change?  I believe because we're adding net NEW we don't need to bump it, but wanted to check with you @mikeshng

Reply:
> Did the AnsibleJob version change? Does the curator version need to change? I believe because we're adding net NEW we don't need to bump it, but wanted to check with you @mikeshng

No, I don't think we don't need to bump the version. I checked upstream AnsibleJob version
https://github.com/ansible/awx-resource-operator/blob/devel/config/crd/bases/tower.ansible.com_ansiblejobs.yaml
and the [inventory PR](https://github.com/stolostron/cluster-curator-controller/pull/119/files) that also added a net new field to CRD.